bindings/go/src/fdb/database.go

```diff
@@ -89,8 +91,10 @@ func retryable(wrapped func() (interface{}, error), onError func(Error) FutureNi
 			return
 		}
 
-		ep, ok := e.(Error)
-		if ok {
+		// Check if the error chain contains an
+		// fdb.Error
+		var ep Error
+		if xerrors.As(e, &ep) {
```

Comment:
I see. So, the FDB error does not implement the `Wrapper` interface as it, itself, has no "cause" error. But if I understand the idea of this PR correctly, then the idea is that someone might have their own error type that wrapped FDB errors (and added more information that the application found useful, let's say). Then they could wrap an FDB error and still have it be retried as long as *that* error implemented the `Wrapper` interface.

That does seem useful, if that's the idea here.

Reply:
This is exactly the idea, users can wrap the fdb error so that it is still retryable. 👍